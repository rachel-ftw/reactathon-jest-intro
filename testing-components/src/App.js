import React, { Component } from 'react'
import Img from './Img'
import Form from './Form'
import './extras/App.css'
import fetchGif from './extras/utils'

const giphyAPIKey = process.env.REACT_APP_GIPHY_API_KEY

class App extends Component {
  state = {
    imageUrl: '',
  }

  handleSubmit = (event, searchTerm) => {
    const url = `http://api.giphy.com/v1/gifs/translate?api_key=${giphyAPIKey}&s=${searchTerm}`
    event.preventDefault()
    fetchGif(url)
      .then(imageUrl => this.setState({ imageUrl }))
  }

  render() {
    const { imageUrl } = this.state

    return (
      <div className="App">
        <Form handleSubmit={this.handleSubmit} />
        <Img imageUrl={imageUrl} />
      </div>
    )
  }
}

export default App
