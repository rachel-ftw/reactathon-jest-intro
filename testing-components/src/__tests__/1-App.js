import React from 'react'
import { shallow } from 'enzyme'
// enzyme docs: https://github.com/airbnb/enzyme

import App from '../App'
import renderer from 'react-test-renderer'
// docs for renderer: https://reactjs.org/docs/test-renderer.html

describe('<App /> Renderer', () => {
  it('with snapshot & renderer', () => {
    const tree = renderer
      .create(<App />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('with enzyme shallow', () => {
    shallow(<App />)
  });
})
