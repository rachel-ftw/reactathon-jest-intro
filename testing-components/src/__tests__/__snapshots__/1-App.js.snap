// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renderer with snapshot & renderer 1`] = `
<div
  className="App"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search for a Gif"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      search
    </button>
  </form>
  <div />
</div>
`;
